dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -317,16 +292,21 @@ private static CompletionsOptions CreateCompletionsOptions(string text, Complete
         return options;
     }
 
-    private static ChatCompletionsOptions CreateChatCompletionsOptions(ChatRequestSettings requestSettings, ChatHistory chat)
+    private static ChatCompletionsOptions CreateChatCompletionsOptions(ChatRequestSettings requestSettings, IEnumerable<ChatMessageBase> chatHistory)
     {
+        if (requestSettings.ResultsPerPrompt is < 1 or > 128)
```

Comment:
It's not clear what is 128 here. Let's put these numbers in separate constant variables with names.